seqr/utils/middleware.py

```diff
@@ -40,3 +41,38 @@ def process_exception(request, exception):
                     exception_json['traceback'] = traceback_message
             return create_json_response(exception_json, status=_get_exception_status_code(exception))
         return None
+
+class LogRequestMiddleware(MiddlewareMixin):
+
+    @staticmethod
+    def process_response(request, response):
+        # conforms to the httpRequest json spec for stackdriver: https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#HttpRequest
+        http_json = {
+            'requestMethod': request.method,
+            'requestUrl': request.get_raw_uri(),
+            'status': response.status_code,
+            'responseSize': len(response.content) if hasattr(response, 'content') else request.META.get('CONTENT_LENGTH'),
+            'userAgent': request.META.get('HTTP_USER_AGENT'),
+            'remoteIp': request.META.get('REMOTE_ADDR'),
+            'referer': request.META.get('HTTP_REFERER'),
+            'protocol': request.META.get('SERVER_PROTOCOL'),
+        }
+        request_body = None
+        try:
+            if request.body:
+                request_body = json.loads(request.body)
+                # TODO update settings in stackdriver so this isn't neccessary
+                if 'password' in request_body:
+                    request_body['password'] = '***'
```

Comment:
Codacy found an issue: [Possible hardcoded password: '***'](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=6404089)

Reply:
this is okay to ignore